core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/handler/DeleteVolumesValidationHandler.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    @Override
+    public Selectable doAccept(HandlerEvent<DeleteVolumesValidationRequest> deleteVolumesValidationEvent) {
+        LOGGER.debug("Received event: {}", deleteVolumesValidationEvent);
+        DeleteVolumesValidationRequest payload = deleteVolumesValidationEvent.getData();
+        StackDeleteVolumesRequest stackDeleteVolumesRequest = payload.getStackDeleteVolumesRequest();
+        StackDto stack = stackDtoService.getById(payload.getResourceId());
+        String requestGroup = stackDeleteVolumesRequest.getGroup();
+        String blueprintText = stack.getBlueprint().getBlueprintText();
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(blueprintText);
+        Set<String> hostTemplateComponents = processor.getComponentsInHostGroup(requestGroup);
+        Set<ServiceComponent> hostTemplateServiceComponents = processor.getServiceComponentsByHostGroup().get(requestGroup);
+        boolean computeInstance = true;
+        for (String service : hostTemplateComponents) {
+            com.google.common.base.Optional<BlackListedDeleteVolumesRole> enumValue =
+                    Enums.getIfPresent(BlackListedDeleteVolumesRole.class, service);
+            if (enumValue.isPresent()) {
+                computeInstance = false;
+            }
+        }
+        if (!computeInstance) {
+            LOGGER.error("Deleting volumes flow is being stopped, as instance being scaled is not a compute instance.");
+            return new DeleteVolumesResult(
+                    "BadRequestException: Instance group being scaled isn't a compute instance",
+                    new BadRequestException("BadRequestException: Instance group being scaled isn't a compute instance"),
+                    stack.getId());
+//            eventBus.notify(CloudPlatformResult.failureSelector(DeleteVolumesResult.class), new Event<>(deleteVolumesValidationEvent.getHeaders(),
+//                    deleteVolumesResult));
+        } else {
+            List<CloudResource> cloudResourcesToBeDeleted = stack.getResources().stream().filter(resource -> null != resource.getInstanceGroup()
+                            && resource.getInstanceGroup().equals(requestGroup)
+                            && resource.getResourceType().equals(ResourceType.AWS_VOLUMESET))
+                    .map(s -> cloudResourceConverter.convert(s)).collect(toList());
+            String cloudPlatform = stack.getCloudPlatform();
+            try {
+                deleteVolumesService.stopClouderaManagerService(stack, hostTemplateServiceComponents);
+                return new DeleteVolumesRequest(cloudResourcesToBeDeleted, stackDeleteVolumesRequest, cloudPlatform,
+                        hostTemplateServiceComponents);
+            } catch (Exception ex) {
+                LOGGER.error("Deleting volumes flow is being stopped, Exception while stopping CM services.");
+                return new DeleteVolumesResult(
```

Comment:
Can we rename it for DeleteVolumeFailedEvent as it is used from multiple places?